### DISCUSSION TITLE
How does SQLAlchemy update ORM objects?

### DISCUSSION THREAD

**Discussion by minh-ht** [sqlalchemy/sqlalchemy#8562] [Usage Questions] [answered]
I find that if I have multiple ORM objects of the same object in database and I update one object by changing the attribute and commit, all of them get updated. How does it work? I used to use `session.refresh(object)` to have one updated.

Below is an example:

```python

object_1 = await get_object(session, id)
object_2 = await get_object(session, id)

print(object_2.name)    # Print "Name"

object_1.name = "Changed name"
await session.commit()

print(object_2.name)    # Print "Changed name"

```

**Comment by CaselIT** [comment, marked as answer]
Hi,

These are not multiple object, but the same one. This assertion will pass: `assert object_1 is object_2`. the session keeps a single instance for each id so the second get will return the same object loaded by the first get

**Reply by minh-ht** [reply]
Thank you so much. Now I understand how it works.
